feat(build): Use code splitting to dedupe main & core entries

Andarist opened the pull request:
**What**:

Share common code between main and core entries

**Why**:

Dedupe distributed code.

**How**:

Leveraging rollup's code splitting feature (used by newest kcd-scripts)

**Checklist**:
- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->
- [ ] Added myself to contributors table <!-- this is optional, see the contributing guidelines for instructions -->

**NOTE**
There is still 1 issue with this PR - I'm not sure where `core`'s typings should live. Are you familiar with resolution algorithm of those?

Andarist commented on core/package.json:
```diff
@@ -0,0 +1,5 @@
+{
```
this one is auto-generated by kcd-scripts, maybe u'd like to `.gitignore` it?

Andarist commented:
cc @alitaheri

alitaheri commented:
@Andarist This library doesn't really have typings since it's api surface is very small. Besides, typescript looks for the `typings` prop in the package.json file if it can't resolve the typings using it's resolution mechanism. so, as long as the package.json has the props `types` referring to `types.d.ts` everything is good 👍 

Since changes by the way 👍

alitaheri commented:
As long as the reference is valid. Simply copy the types.d.ts file and make sure the nested package.json has the types prop. All shoul then be fine

Andarist commented:
@kentcdodds we'd have to introduce `postbuild` script that would extend generated `package.json` with `typings` field. Let me know if I should add it or maybe there is some other option to support it?

Andarist commented:
Ok, I've added a script to add those typing fields to extra `package.json`s. I've also tweaked docs regarding importing `/core` and added a deprecation warning in `core.esm.js`.

Andarist commented on README.md:
```diff
@@ -1,3 +1,24 @@
+
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+* [rtl-css-js](#rtl-css-js)
```
this had to be added by some script

kentcdodds replied:
Yup. Could you move this down to below "This solution?" Then it will be regenerated to only include stuff below it. Also add a `## Table of Contents` above it as well.

Andarist commented on README.md:
```diff
@@ -123,8 +144,11 @@ I'm not aware of any, if you are please [make a pull request](http://makeapullre
-| [<img src="https://avatars.githubusercontent.com/u/1500684?v=3" width="100px;"/><br /><sub>Kent C. Dodds</sub>](https://kentcdodds.com)<br />[💻](https://github.com/kentcdodds/rtl-css-js/commits?author=kentcdodds "Code") [⚠️](https://github.com/kentcdodds/rtl-css-js/commits?author=kentcdodds "Tests") [🚇](#infra-kentcdodds "Infrastructure (Hosting, Build-Tools, etc)") | [<img src="https://avatars ...
-| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
+
+| [<img src="https://avatars.githubusercontent.com/u/1500684?v=3" width="100px;"/><br /><sub>Kent C. Dodds</sub>](https://kentcdodds.com)<br />[💻](https://github.com/kentcdodds/rtl-css-js/commits?author=kentcdodds 'Code') [⚠️](https://github.com/kentcdodds/rtl-css-js/commits?author=kentcdodds 'Tests') [🚇](#infra-kentcdodds 'Infrastructure (Hosting, Build-Tools, etc)') | [<img src="https://avatars ...
```
this change probably done by prettier

kentcdodds replied:
Hmm... That's weird. all-contributors-cli was updated to add a `prettier-ignore` comment above this. Try running `npx kcd-scripts contributors generate` and it should fix this and add a `prettier-ignore` comment automatically.

Andarist commented on README.md:
```diff
@@ -90,7 +111,7 @@ console.log(styles) // logs {'padding-left': 23}
 yourself. To use these you can use the `rtlCSSJSCore` global with the UMD build,
 `require('rtl-css-js/core')`, or use
-`import {propertyValueConverters, arrayToObject} from 'rtl-css-js/core.esm'`.
+`import {propertyValueConverters, arrayToObject} from 'rtl-css-js/core'`.
```
this is my only change to this file

kentcdodds commented on core.esm.js:
```diff
@@ -1,3 +1,8 @@
+if (process.env.NODE_ENV !== 'production') {
+  // eslint-disable-next-line no-console
+  console.warn(
+    'Importing `rtl-css-js/core.esm` is deprecated, please use `rtl-css-js/core`.',
```
Thanks!

kentcdodds commented on scripts/add-proxies-typings.js:
```diff
@@ -0,0 +1,16 @@
+const fs = require('fs')
```
:+1: This is great

Andarist commented:
Ok, I think this is ready to merge now 😄
